org.jacoco.core.test/src/org/jacoco/core/test/filter/EnumSwitchTest.java

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.test.filter;
+
+import org.jacoco.core.analysis.ICounter;
+import org.jacoco.core.test.filter.targets.EnumSwitch;
+import org.jacoco.core.test.validation.JavaVersion;
+import org.jacoco.core.test.validation.ValidationTestBase;
+import org.junit.Test;
+
+/**
+ * Test of filtering of a synthetic class that is generated by javac for a enum
+ * in switch statement.
+ */
+public class EnumSwitchTest extends ValidationTestBase {
+
+	public EnumSwitchTest() {
+		super(EnumSwitch.class);
+	}
+
+	@Test
+	public void testCoverageResult() {
+		if (isJDKCompiler && JAVA_VERSION.isBefore("1.6")) {
+			assertLine("switch", ICounter.PARTLY_COVERED, 0, 2);
```

Comment:
Should we leave a comment why this is the case?

Reply:
@marchof added comment